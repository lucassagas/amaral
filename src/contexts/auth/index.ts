export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
