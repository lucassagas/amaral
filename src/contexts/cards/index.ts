export { CardContext, CardContextProvider } from './cards';

