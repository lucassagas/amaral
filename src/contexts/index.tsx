import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'
import { AuthContextProvider } from './auth'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <AuthContextProvider>
        <QueryClientProvider client={queryClient}>
          {children}
          <ReactQueryDevtools />
        </QueryClientProvider>
      </AuthContextProvider>
    </ChakraProvider>
  )
}
