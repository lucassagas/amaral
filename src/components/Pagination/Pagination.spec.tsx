import { AppProvider } from '@/contexts/index'
import { render, screen } from '@testing-library/react'
import { Pagination } from './Pagination'

type PaginationProps = {
  currentPage?: number
  totalCountOfRegisters: number
}

const renderComponent = ({
  currentPage,
  totalCountOfRegisters,
}: PaginationProps) => {
  return render(
    <AppProvider>
      <Pagination
        totalCountOfRegisters={totalCountOfRegisters}
        registersPerPage={10}
        currentPage={currentPage}
        onPageChange={() => {}}
      />
    </AppProvider>
  )
}

describe('<Pagination />', () => {
  it('should render correctly', () => {
    const { container } = renderComponent({
      currentPage: 1,
      totalCountOfRegisters: 30,
    })

    expect(container).toMatchSnapshot()
  })

  it('should render with ellipses', () => {
    renderComponent({ currentPage: 15, totalCountOfRegisters: 100 })
    expect(screen.getByText(/\.\.\./i)).toBeInTheDocument()
  })

  it('should render with two ellipses', () => {
    renderComponent({ currentPage: 15, totalCountOfRegisters: 200 })

    expect(screen.getAllByText(/\.\.\./i)[1]).toBeInTheDocument()
  })
})
