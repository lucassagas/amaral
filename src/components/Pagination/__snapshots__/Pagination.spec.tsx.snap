// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render correctly 1`] = `
<div>
  <div
    class="chakra-stack css-1fjzvpp"
  >
    <div
      class="css-0"
    >
      <strong>
        1
      </strong>
       -
       
      <strong>
        10
      </strong>
       
      de 
      <strong>
        30
      </strong>
    </div>
    <div
      class="chakra-stack css-wta9zy"
    >
      <button
        class="chakra-button css-1jw1j19"
        disabled=""
        type="button"
      >
        1
      </button>
      <button
        class="chakra-button css-u30hwm"
        type="button"
      >
        2
      </button>
      <button
        class="chakra-button css-u30hwm"
        type="button"
      >
        3
      </button>
    </div>
  </div>
</div>
`;
